examples/echo_client_with_mam.py

```diff
@@ -0,0 +1,206 @@
+                                           #start="2017-06-05T12:00:00+00:00",
+                                           #end="2017-06-05T23:59:59+00:00",
+                                           number_of_queried_elements=1,
+                                           collect_all=True)
+        
+        ## If no callback is used, the handler function is called here to display the results.
+        self.__handle_mam_result(answer)
+        
+        ## Note that a blocking call will finish before this,
+        #  an async call will finish after this.
+        print("End MAM")
+        
+        
+    def __handle_mam_result(self, response):
+        print( "Answer for query {}".format(response['mam_answer']['queryid']) )
+        print( "  - answer is: {}".format("complete" if response["mam_answer"]["complete"] else "incomplete") )
+        print( "  - last message is: {}".format(response["mam_answer"]["rsm"]["last"]) )
+        
+        result = response['mam_answer']['results']
+        
+        for x in result:
+            msg = x["mam_result"]["forwarded"]["message"]
+            delay = x["mam_result"]["forwarded"]["delay"]
+            if ( msg["body"] ):
+                print("==========")
+                print( "From:", msg["from"] )
+                print( "Time:", delay["stamp"] )
+                print( "--------" )
+                print( msg["body"] )
+                print( "--------" )
+                print()
+
+
+    def message(self, msg):
+        """
+        Process incoming message stanzas. Be aware that this also
+        includes MUC messages and error messages. It is usually
+        a good idea to check the messages's type before processing
+        or sending replies.
+
+        Arguments:
+            msg -- The received message stanza. See the documentation
+                   for stanza objects and the Message stanza to see
+                   how it may be used.
+        """
+        if msg['type'] in ('chat', 'normal'):
+            msg.reply("Thanks for sending\n%(body)s" % msg).send()
+
+
+if __name__ == '__main__':
+    # Setup the command line arguments.
+    optp = OptionParser()
+
+    # Output verbosity options.
+    optp.add_option('-q', '--quiet', help='set logging to ERROR',
+                    action='store_const', dest='loglevel',
+                    const=logging.ERROR, default=logging.INFO)
+    optp.add_option('-d', '--debug', help='set logging to DEBUG',
+                    action='store_const', dest='loglevel',
+                    const=logging.DEBUG, default=logging.INFO)
+    optp.add_option('-v', '--verbose', help='set logging to COMM',
+                    action='store_const', dest='loglevel',
+                    const=5, default=logging.INFO)
+
+    # JID and password options.
+    optp.add_option("-j", "--jid", dest="jid",
+                    help="JID to use")
+    optp.add_option("-p", "--password", dest="password",
+                    help="password to use")
```

Comment:
I thought about this, but I the think the example focuses on developers who want to get into the code and need a point to start. Therefore, having lots of options from a users perspective is no priority.